### PR TITLE
Add an option in local model trainer to remove inputs and container artifacts

### PR DESCRIPTION
**Pull request by pintaoz-aws**
…using local model trainer

*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)
- [ ] If adding any dependency in requirements.txt files, I have spell checked and ensured they exist in PyPi

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/modules/train/model_trainer.py [hunk 1]
```diff
@@ -203,6 +203,8 @@ class ModelTrainer(BaseModel):
         local_container_root (Optional[str]):
             The local root directory to store artifacts from a training job launched in
             "LOCAL_CONTAINER" mode.
+        remove_inputs_and_container_artifacts (Optional[bool]):
```
Comment (mufaddal-rohawala) [inline src/sagemaker/modules/train/model_trainer.py:206]: This option seems to be specifically for local mode? Can we not just do the cleanup for local mode by default?
We should be very careful in expanding the input param list for ModelTrainer (we shouldn't unless User experience warrants it).

**Review by mufaddal-rohawala — CHANGES_REQUESTED** [review]
https://github.com/aws/sagemaker-python-sdk/pull/4978/files#r1898041392

File: src/sagemaker/modules/local_core/local_container.py [hunk 2]
```diff
@@ -108,6 +108,8 @@ class _LocalContainer(BaseModel):
     container_entrypoint: Optional[List[str]]
     container_arguments: Optional[List[str]]
 
+    _temperary_folders: List[str] = []
```
Comment (nargokul) [inline src/sagemaker/modules/local_core/local_container.py:111]: Nit: `temporary_folders`

File: src/sagemaker/modules/train/model_trainer.py (hunk 1 above)
Comment (nargokul) [reply src/sagemaker/modules/train/model_trainer.py:206]: +1

Is there any advantage we provide by keeping the artifacts stored?

**Review by nargokul — CHANGES_REQUESTED** [review]

File: src/sagemaker/modules/train/model_trainer.py (hunk 1 above)
Comment (pintaoz-aws) [reply src/sagemaker/modules/train/model_trainer.py:206]: the users may need to check the environment variables and hyperparameters in each container to debug

**Review by mufaddal-rohawala — APPROVED** [review]
